rfc/rfc-76/rfc-76.md

```diff
@@ -0,0 +1,143 @@
+-->
+# RFC-[76]: [support EventTimeBasedCompactionStrategy and metric for RO table]
+
+## Proposers
+
+- @waitingF
+
+## Approvers
+ - @danny0405
+ - @voonhous
+
+## Status
+
+JIRA: [HUDI-6979](https://issues.apache.org/jira/browse/HUDI-6979)
+
+> Please keep the status updated in `rfc/README.md`.
+
+## Abstract
+
+Currently, to gain low ingestion latency, we can adopt the MergeOnRead table, which support appending log files and 
+compact log files into base file later. When querying the snapshot table (RT table) generated by MOR, 
+query side have to perform a compaction so that they can get all data, which is expected time-consuming causing query latency.
+At the time, hudi provide read-optimized table (RO table) for low query latency just like COW.
+
+Generally the data to ingest arrives roughly in event time order. For cases some users read the hudi table,
+they may not concern with the immediate full data, but the data before a specified time T (eg. data before 0 clock).
+For such cases, user want all data before a specified time, user have to query the RT table to get all data with expected high query latency.
+
+Based on this, we want to implement the strategy based on the event time: `EventTimeBasedCompactionStrategy`. 
+
+
+## Background
+
+Currently, there is no compaction strategy based on event time, the only `DayBasedCompactionStrategy` need the table partitioned by day in specified format(yyyy/mm/dd), 
+which is not very general and has too large a time granularity. There is no data completion/freshness guarantee or metrics for RO table.
+
+Based on this, we plan to 
+1. launch a compaction strategy based on event time: `EventTimeBasedCompactionStrategy`.
+2. report RO table data completion and freshness.
+
+With the data completion and freshness of RO table, we can expand use-case of RO table.
+That is given event time threshold T, the log files before T can be compacted, then resulting RO table obtains all data before T, with low query latency.
+
+Currently, we introduced new file slicing algorithm based on barrier and generating compaction plan based on completion time [[HUDI-6495][RFC-66]](https://github.com/apache/hudi/pull/7907).
+
+With the barrier file slicing, file slices are split by the barriers extracted from every instant time of base files 
+or the first instant of log file if no base file exists.
+
+When generating compaction plan, the scheduler will filter all log files with completion time before the `compactionInstant`. 
+And the compaction operation will generate a new base file with `compactionInstant` as its instant time;
+
+So the file slices are split naturally by the `compactionInstant` for MOR cases. 
+
+So we can replace the `compactionInstant` with `eventTimeInstant` computed by `EventTimeBasedCompactionStrategy` when generating plan. 
+In this case, file slices are split by `eventTimeInstant` instead of `compactionInstant`. 
+This is the base for `EventTimeBasedCompactionStrategy`.
```

Comment:
feasible, just to note that the generation of the `eventTImeInstant` must share the same lock with normal instant time generation.

Reply:
sure, noted